infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -54,12 +54,27 @@ public void write(@Nonnull K key, @Nonnull V value) {
         try {
             CompletableFuture<Void> cf = new CompletableFuture();
             operationsQueue.add(new BatchWriterOperation(BatchWriterOperation.Type.WRITE,
-                    (Long) key, (LogData) value, cf));
+                    (Long) key, (LogData) value, null, cf));
             cf.get();
         } catch (Exception e) {
             log.trace("Write Exception {}", e);
             if (e.getCause() instanceof RuntimeException) {
-                throw (RuntimeException)e.getCause();
+                throw (RuntimeException) e.getCause();
+            } else {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void bulkWrite(List<LogData> entries) {
+        try {
+            CompletableFuture<Void> cf = new CompletableFuture();
+            operationsQueue.add(new BatchWriterOperation(BatchWriterOperation.Type.RANGE_WRITE,
+                    null, null, entries, cf));
+        } catch (Exception e) {
+            log.trace("Write Exception {}", e);
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
             } else {
                 throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)